week-9/Homework/mandatory/2-exercises/4-music-player.js

```diff
@@ -20,18 +21,52 @@ This means the order the songs are played in will be random, but each song will
 
 
 class MusicPlayer {
+
     // Add your code here
+    constructor() {
+        this.allSongs = [];
+        this.counter = 0;
+    }
+    add(song, artist) {
+
+        this.allSongs.push(`${song} by ${artist}`);
+    }
+    play() {
+        if (this.allSongs.length===0) {
+            console.log('You do not have any songs to play');
+        }
+        else {
+            console.log(`Currently playing ${this.allSongs[this.counter]} `);
+        }
+        
+    }
+    skip() {
+        this.counter += 1;
```

Comment:
What happens if the user keeps on skipping in the tracklist -  your function eventually ends up printing `Currently playing undefined`

Reply:
Think about how you could manage the value of the counter when you reach the end of the array